src/3box.js

```diff
@@ -57,91 +56,61 @@ class ThreeBox {
+    console.timeEnd('init stores')
+
+    this._createRootStore(rootStoreAddress, privOdbAddress, pubOdbAddress, pinningNode)
+
+    return {rootStoreAddress, privOdbAddress, pubOdbAddress}
+  }
+
+  async _sync (privOdbAddress, pubOdbAddress) {
+    console.time('sync stores')
+    await Promise.all([this.public._sync(pubOdbAddress), this.private._sync(privOdbAddress)])
+    console.timeEnd('sync stores')
+  }
+
+  async _createRootStore(rootStoreAddress, privOdbAddress, pubOdbAddress, pinningNode) {
+    console.time('opening pinning room, pinning node joined')
+    this._pubsub = new Pubsub(this._ipfs, (await this._ipfs.id()).id)
+    const onNewPeer = (topic, peer) => {
+      console.log('Peer joined the room', peer)
+      console.log(peer, pinningNode.split('/').pop())
+      if (peer === pinningNode.split('/').pop()) {
+        console.timeEnd('opening pinning room, pinning node joined')
+        console.log('broadcasting odb-address')
+        this._pubsub.publish(PINNING_ROOM, {type: 'PIN_DB', odbAddress: rootStoreAddress })
       }
-      await Promise.all(storePromises)
-      console.timeEnd('sync stores')
-    } else {
-      const rootStoreName = didFingerprint + '.root'
-      console.time('create rootStore')
-      this._rootStore = await this._orbitdb.feed(rootStoreName)
-      console.timeEnd('create rootStore')
-      console.time('add PublicStore to rootStore')
-      await this._rootStore.add({ odbAddress: await this.public._sync() })
-      console.timeEnd('add PublicStore to rootStore')
-      console.time('add PrivateStore to rootStore')
-      await this._rootStore.add({ odbAddress: await this.private._sync() })
-      console.timeEnd('add PrivateStore to rootStore')
-      rootStoreAddress = this._rootStore.address.toString()
-      console.time('publish rootStoreAddress to address-server')
-      this._publishRootStore(rootStoreAddress)
-      console.timeEnd('publish rootStoreAddress to address-server')
-      console.time('broadcast rootStoreAddress over pubsub')
-      this._pubsub.subscribe(PINNING_ROOM, () => {}, onNewPeer)
-      console.timeEnd('broadcast rootStoreAddress over pubsub')
     }
+    console.time('open rootStore')
+    this._rootStore = await this._orbitdb.open(rootStoreAddress)
+    console.timeEnd('open rootStore')
+    console.time('add PublicStore to rootStore')
+    await this._rootStore.add({ odbAddress: pubOdbAddress })
+    console.timeEnd('add PublicStore to rootStore')
+    console.time('add PrivateStore to rootStore')
+    await this._rootStore.add({ odbAddress: privOdbAddress })
+    console.timeEnd('add PrivateStore to rootStore')
+    console.time('publish rootStoreAddress to address-server')
+    this._publishRootStore(rootStoreAddress)
+    console.timeEnd('publish rootStoreAddress to address-server')
+    console.time('broadcast rootStoreAddress over pubsub')
+    this._pubsub.subscribe(PINNING_ROOM, () => {}, onNewPeer)
```

Comment:
It might be a good idea to do this on line 99 or so. It isn't blocking anything else, and if it happens earlier the syncs will be faster!